docs/content/guides/developer/app-examples/plinko.mdx

```diff
@@ -0,0 +1,743 @@
+        game.player
+    }
+
+    /// Returns the player's vrf_input bytes.
+    public fun vrf_input(game: &Game): vector<u8> {
+        game.vrf_input
+    }
+
+    /// Returns the fee of the game.
+    public fun fee_in_bp(game: &Game): u16 {
+        game.fee_bp
+    }
+
+    // === Admin Functions ===
+
+    /// Helper function to calculate the amount of fees to be paid.
+    /// Fees are only applied on the player's stake.
+    public fun fee_amount(game_stake: u64, fee_in_bp: u16): u64 {
+        ((((game_stake / (GAME_RETURN as u64)) as u128) * (fee_in_bp as u128) / 10_000) as u64)
+    }
+
+    /// Helper function to check if a game exists.
+    public fun game_exists(house_data: &HouseData, game_id: ID): bool {
+        dof::exists_(hd::borrow(house_data), game_id)
+    }
+
+    /// Helper function to check that a game exists and return a reference to the game Object.
+    /// Can be used in combination with any accessor to retrieve the desired game field.
+    public fun borrow_game(game_id: ID, house_data: &HouseData): &Game {
+        assert!(game_exists(house_data, game_id), EGameDoesNotExist);
+        dof::borrow(hd::borrow(house_data), game_id)
+    }
+
+    // === Private Functions ===
+
+    /// Internal helper function ussed to create a new game. 
+    /// The player must provide a guess and a Counter NFT.
+    /// Stake is taken from the player's coin and added to the game's stake. 
+    /// The house's stake is also added to the game's stake.
+    fun internal_start_game(counter: &mut Counter, num_balls: u64, coin: Coin<SUI>, house_data: &HouseData, fee_bp: u16, ctx: &mut TxContext): (ID, Game) {
+        // Ensure guess is valid.
+        // map_guess(guess);
+        let user_stake = coin::value(&coin);
+        // Ensure that the stake is not higher than the max stake.
+        assert!(user_stake <= hd::max_stake(house_data), EStakeTooHigh);
+        // Ensure that the stake is not lower than the min stake.
+        assert!(user_stake >= hd::min_stake(house_data), EStakeTooLow);
+        // Ensure that the house has enough balance to play for this game.
+        assert!(hd::balance(house_data) >= (user_stake*(*vector::borrow(&hd::multiplier(house_data), 0)))/100, EInsufficientHouseBalance);
+
+        let vrf_input = counter_nft::get_vrf_input_and_increment(counter, num_balls);
+
+        let id = object::new(ctx);
+        let game_id = object::uid_to_inner(&id);
+
+        let new_game = Game {
+            id,
+            game_start_epoch: tx_context::epoch(ctx),
+            stake: coin::into_balance<SUI>(coin),
+            player: tx_context::sender(ctx),
+            vrf_input,
+            fee_bp
+        };
+
+        emit(NewGame {
+            game_id,
+            player: tx_context::sender(ctx),
+            vrf_input,
+            user_stake,
+            fee_bp
+        });
+
+        (game_id, new_game)
+    }
+
+}
+```
+
+#### Constants
+
+- `GAME_RETURN`: Defines the return multiplier for the game.
```

Comment:
This is not the game multiplier; it's used in fee calculations. It's still to be determined if we will keep fees in the contract. We might remove them.